engine/src/test/java/io/camunda/zeebe/engine/processing/bpmn/multiinstance/MultiInstanceActivityTest.java

```diff
@@ -894,6 +894,57 @@ public void shouldSetOutputElementVariable() {
             OUTPUT_COLLECTION.stream().map(JsonUtil::toJson).collect(Collectors.toList()));
   }
 
+  @Test
+  public void shouldNotInitializeOutputElementVariableIfSameNameAsInputElement() {
```

Comment:
Good question. 

This was somewhat discussed in another [issue](https://github.com/camunda/zeebe/issues/7330#issuecomment-865968125). To summarize, the input collection variable exists at a higher scope, the output collection variable is created at the multi-instance element's scope. So the output collection variable can shadow the input collection variable. 

What that analysis doesn't discuss is the case combined with an input mapping to set the input collection variable at the multi-instance element scope. However, that shouldn't work because the input mappings are only applied for each inner activity, not for the entire multi-instance element. As far as I know, we consider it not a valid use case at this time.